deployments/ansible/roles/signalfx-agent/tasks/main.yml

```diff
@@ -0,0 +1,15 @@
+---
+# Check Signalfx Access Toekn is defined
+- name: Check SignalFx Access Token is defined
+  fail: msg="Bailing out. This play requires 'signalFxAccessToken'"
+  when: conf['signalFxAccessToken'] is undefined or conf['signalFxAccessToken'] is none or (conf['signalFxAccessToken'] | trim == '')
+
+# Import tasks according to the distribution
+- name: Import signalfx-agent deploy for CentOS or RHEL
+  import_tasks: yum_repo.yml
+  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux' or ansible_distribution == 'Amazon'
+
+- name: Import signalfx-agent deploy for Debian or Ubuntu
```

Comment:
Have it throw an error if the distribution is neither centos or debian based (e.g. Alpine, ArchLinux, etc.).

Reply:
No, it will not throw any error. It will just skip the tasks if the condition not matched. Going to add the 'conditional' with appropriate message when the play is performed on other distributions!